src/reactors/PriorityOrderReactorQuoter.sol

```diff
@@ -0,0 +1,142 @@
+
+            auctionStartBlock = order.cosignerData.auctionTargetBlock;
+        }
+
+        resolvedOrder = ResolvedOrder({
+            info: order.info,
+            input: order.input.scale(priorityFee),
+            outputs: order.outputs.scale(priorityFee),
+            sig: abi.encode(signedOrder.sig, auctionStartBlock),
+            hash: orderHash
+        });
+    }
+
+    /// @inheritdoc BaseReactor
+    function _transferInputTokens(ResolvedOrder memory order, address to) internal override {
+        (bytes memory sig, uint256 auctionStartBlock) = abi.decode(order.sig, (bytes, uint256));
+        permit2.permitWitnessTransferFrom(
+            order.toPermit(),
+            order.transferDetails(to),
+            order.info.swapper,
+            order.hash,
+            PriorityOrderLib.PERMIT2_ORDER_TYPE,
+            sig
+        );
+        /// revert if the resolved auctionStartBlock is in the future
+        if (block.number < auctionStartBlock) {
+            revert OrderNotFillable();
+        }
+    }
+
+    /// @notice validate the priority order fields
+    /// - deadline must be in the future
+    /// - resolved auctionStartBlock must not be in the future
+    /// - if input scales with priority fee, outputs must not scale
+    /// @dev Throws if the order is invalid
+    function _validateOrder(bytes32 orderHash, PriorityOrder memory order) internal view {
+        if (order.info.deadline < block.timestamp) {
+            revert InvalidDeadline();
+        }
+
+        if (order.input.mpsPerPriorityFeeWei > 0) {
+            for (uint256 i = 0; i < order.outputs.length; i++) {
+                if (order.outputs[i].mpsPerPriorityFeeWei > 0) {
+                    revert InputOutputScaling();
+                }
+            }
+        }
+    }
+
+    /// @notice resolve the priority fee for the current transaction
+    /// @notice tx.gasprice must be greater than or equal to block.basefee
+    /// @param baselinePriorityFeeWei the baseline priority fee to be subtracted from calculated priority fee
+    /// @return priorityFee the resolved priority fee
+    function _getPriorityFee(uint256 baselinePriorityFeeWei) internal view returns (uint256 priorityFee) {
+        if (tx.gasprice < block.basefee) revert InvalidGasPrice();
+        unchecked {
+            priorityFee = tx.gasprice - block.basefee;
+            if (priorityFee > baselinePriorityFeeWei) {
+                priorityFee -= baselinePriorityFeeWei;
+            } else {
+                priorityFee = 0;
+            }
+        }
+    }
+
+    /// @notice check if an order has already been filled
+    /// @dev implementation copied from https://github.com/Uniswap/permit2/blob/cc56ad0f3439c502c246fc5cfcc3db92bb8b7219/src/SignatureTransfer.sol#L150
+    /// @param swapper the address of the swapper
+    /// @param nonce the nonce associated with the order
+    function _checkPermit2Nonce(address swapper, uint256 nonce) internal view {
```

Comment:
maybe worth moving this to a library, now that it's used in both the reactor and quoter?